app-experimental.tex

```diff
@@ -2395,3 +2395,219 @@ \section{Distributed Capabilities For Peripherals And Accelerators} % <<<
+Please note that every capability within a compartment can be manipulated by the compartment itself.
+This also includes CID sealed code capabilities.
+In the CHERI world up to now, sentries cannot be manipulated.
+
+At the moment, this proposal is limited to the 128-bit capability format.
+However, the underlying mechanism to add CID bits to the meta information bits of the capability works for every format.
+In the 128-bit capability format, we propose to allocate 10 bits to the compartment ID, but we envision any number of bits fitting in the meta  infomration bits to a valid implementation of CID Sealing.
+This length is also known as the ACID\_LENGTH, and keep the remaining 8 bits reserved, which allows for 1024 CIDs to be encoded.
+These CIDs are referred to as the Architectural CIDs (ACIDs).
+Software may choose to virtualise these and can create Software CIDs (SCIDs), which are a concept similar to PIDs.
+A SCID may consist of the corresponding ACID added to other bits in order to create a virtual identifier or can be fully independent of the ACID is mapped to.
+
+Compartment IDs come with two new instructions.
+One instruction for reading the CID into a general purpose register and one instruction for setting the CID of a capability.
+Like capabilities themselves, CIDs are not considered secret.
+Thereforem, the CID reading instruction is not privileged, but the writing instruction is restricted (see below).
+We suggest encodings for the instructions in order to demonstrate the adaptability of CID to the currently existing CHERI-RISC-V ISA.
+
+\texttt{CSetCID cd, rs0}:\\
+set the CID in cd to the value in rs0.
+This instruction needs the PERMIT\_SET\_CID bit set, otherwise it throws an exception.
+This instruction uses the ACID\_LENGTH lower bits of rs0 and ignores the upper bits.
+
+Possible encoding (31:0): 0x7f, 0x19, rs0, 0x0, cd, 0x5b (assign a random free funct5)
+
+\texttt{CGetCID rd, cs1}:\\
+extract the CID of cs1 and store it in rd
+
+Preliminary encoding (31:0): 0x7f, 0x1, cs1, 0x0, rd, 0x5b (this is the same encoding as CGetType, which will be obsolete with this proposal)
+
+The PERMIT\_SET\_CID bit is a hardware permission bit that, if set in a PCC, allows manipulating the ACID.
+Like all hardware permission bits in CHERI, PERMIT\_SET\_CID is constrained by monotonicity.
+A code capability with this bit set should only be available to supervisor code.
+
+
+\subsection{Sealing}
+
+The current CID (the CID of the current compartment) is determined by the CID in the PCC, which is also a capability.
+If the PCC changes and the new PCC has a different CID, this constitutes a compartment change (see Section~\ref{subsec:comp_change}).
+
+With an ACID being in all capabilities, we can establish a concept referred to as CID Sealing.
+All capabilities are implicitly sealed by their CID.
+This means that they can only be unsealed if their CID matches the \texttt{PCC.CID}.
```

Comment:
you mentionned a few paragraphs ago
> All values but the zero CID (CID==0) are valid IDs.

I presume a value of 0 in the CID field means "always unsealed" ?
If so, I would slightly rephrase that sentence with something along the lines of:

"Capabilities are considered unsealed if and only if: their CID matches PCC.CID or their CID is 0. All other capabilities are implicitly sealed."